bot/cogs/birthday/birthday_commands.py

```diff
@@ -0,0 +1,118 @@
+from datetime import datetime, timezone, time
+import asyncio
+
+from nextcord import slash_command, Interaction, SlashOption, Permissions, Member, TextChannel
+from nextcord.ext import commands, tasks
+
+from bot.config import Config
+from bot.utils import messages
+from bot.utils.constants import TESTING_GUILD_ID, BUMPERS_GUILD_ID
+from db.helpers import birthday_helper
+
+class BirthdayCommands(commands.Cog):
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.post_birthdays.start()
+    
+    # Every day at 9AM Pacific Time
+    @tasks.loop(time=time(hour=17, minute=0, second=0, tzinfo=timezone.utc))
+    async def post_birthdays(self):
+        if not self.bot.is_ready():
+            return
+
+        birthdays = birthday_helper.get_todays_birthdays()
+        current_guild_id = None
+        birthday_channel_id = None
+        message = None
+        guild = None
+        for birthday in birthdays:
+            if birthday.guild_id != current_guild_id:
+                if message is not None and birthday_channel_id is not None:
+                    message = messages.get_special_birthday_fields(message)
+                    channel = self.bot.get_channel(birthday_channel_id)
+                    await channel.send(embed=message)
+                    message = None
+                guild = self.bot.get_guild(birthday.guild_id)
+                if guild is None:
+                    continue
+                current_guild_id = birthday.guild_id
+                birthday_channel_id = birthday_helper.get_birthday_channel_id(current_guild_id)
+                if birthday_channel_id is None:
+                    continue
+                message = messages.birthday_message()
+                member = guild.get_member(birthday.user_id)
+                message = messages.birthday_entry(message, birthday, member)
+            else:
+                member = guild.get_member(birthday.user_id)
+                message = messages.birthday_entry(message, birthday, member)
+        if message is not None and birthday_channel_id is not None: 
+            message = messages.get_special_birthday_fields(message)
+            channel = self.bot.get_channel(birthday_channel_id)
+            await channel.send(embed=message)
+
+
+    @post_birthdays.error
```

Comment:
Definitely want it if only to restart the task on error. Without it, if there were an error the task would die until the bot were restarted 👍 

and even if the code is flawless, there's always an error at some point because Discord will always go down at some point, lol

Reply:
Yeah that's fair, I'm just not sure if it's a great idea to just sleep for a minute and then restart the task? In case that error is still there once the task is restarted? Doesn't hurt to keep it anyway I suppose :)